brainfuck/commands.go

```diff
@@ -0,0 +1,52 @@
+package brainfuck
+
+import "fmt"
+
+type command interface {
+	Execute(mem *memory)
+}
+
+type MoveForward struct{}
+
+func (MoveForward) Execute(mem *memory) {
+	if mem.ptr == len(mem.buffer)-1 {
```

Comment:
didn't get an idea.. why is this if condition needed?

Reply:
Fixed